conans/client/conan_api.py

```diff
@@ -681,6 +683,78 @@ def export_alias(self, reference, target_reference):
         target_reference = ConanFileReference.loads(str(target_reference))
         return self._manager.export_alias(reference, target_reference)
 
+    @api_method
+    def execute(self, command, references=None, profile_name=None,
+                initial_env=None, settings=None, options=None, remote=None,
+                build_modes=None, update=None, env=None, cwd=None):
+        """ Executes a command in the environment defined by the profile, the
+        specified references, settings, and options.
+
+        If any of the required packages is missing they are automatically
+        installed.
+        """
+        # setup proper defaults
+        cwd = cwd if cwd is not None else os.getcwd()
+        references = references if references is not None else []
+        new_env = initial_env if initial_env is not None else os.environ
+
+        # Load the profile given the specified settings, options, and env variables.
+        profile = profile_from_args(profile_name, settings, options, env, cwd,
+                                    self._client_cache)
+
+        # Build a temporary conanfile.txt, install it and determine the
+        # required environment variables. Note that, as long as extension
+        # is correct, the actual name does not matter.
+        env_vars = None
+
+        # Write all requested references to the file. Note that we have to
+        # close the file such that Windows can work with them.
+        # (see https://bugs.python.org/issue14243)
+        temp = tempfile.NamedTemporaryFile(suffix=".txt", delete=False)
+        temp.write(to_file_bytes('[requires]\n'))
+        for ref in references:
+            temp.write(to_file_bytes(ref + '\n'))
+        temp.close()
+
+        # Install the dependencies if needed and get an valid conanfile.
+        conanfile = self._manager.install(reference=temp.name,
+                                            install_folder=None,
+                                            profile=profile,
+                                            remote=remote,
+                                            build_modes=build_modes,
+                                            update=update)
+
+        # Determine the environment variables for the given conanfile.
+        env_vars = VirtualEnvGenerator(conanfile).get_setenv_variables()
+
+        # Delete the temporary file after the processing.
+        os.unlink(temp.name)
+
+        # Update the environment with the information from the
+        # virtual environment generator.
+        for variable in env_vars:
+            name = variable['name']
+            value = variable['value']
+            if 'separator' in variable.keys() and name in new_env:
```

Comment:
what's the magic separator environment variable?

Reply:
It is simply the separator character which, if it is defined, has to be used to join the current value from the environment with the values from the virtual environment generator. In most cases it is simply `os.pathsep`. However, there are [some variables like CPPFLAGS](https://github.com/niosHD/conan/blob/develop/conans/client/generators/virtualenv.py#L8) that are joined with spaces by the virtual environment generator.